hibernate-core/src/main/java/org/hibernate/type/OneToOneType.java

```diff
@@ -156,7 +178,7 @@ public Object hydrate(
 
 	@Override
 	public boolean isNullable() {
-		return foreignKeyType==ForeignKeyDirection.TO_PARENT;
+		return !constrained;
```

Comment:
@JWGmeligMeyling, I did it that way because, for true one-to-one associations, `ignoreNotFound` effectively means it is not `constrained`.

If the change is removed from the `SecondPass`, then a lot of tests will fail.